icu4c/source/i18n/units_complexconverter.h

```diff
@@ -48,17 +48,33 @@ namespace units {
  */
 class U_I18N_API ComplexUnitsConverter : public UMemory {
   public:
+    /**
+     * Constructs `ComplexUnitsConverter` for an `targetUnit` that could be Single, Compound or Mixed.
+     * In case of:
+     * 1- Single and Compound units,
+     * the conversion will not perform anything, the input will be equal to the output.
+     * 2- Mixed Unit
+     * the conversion will consider the input in the biggest unit. and will convert it to be spread throw
+     * the input units. For example: if input unit is "inch-and-foot", and the input is 2.5. The
+     * converter will consider the input value in "foot", because foot is the biggest unit. Then, it will
+     * convert 2.5 feet to "inch-and-foot".
+     *
+     * @param targetUnit represents the input unit. could be any type. (single, compound or mixed).     *
+     * @param status
+     */
+    ComplexUnitsConverter(const MeasureUnitImpl &targetUnit, const ConversionRates &ratesInfo,
+                          UErrorCode &status);
     /**
      * Constructor of `ComplexUnitsConverter`.
      * NOTE:
-     *   - inputUnit and outputUnits must be under the same category
+     *   - targetUnit and outputUnits must be under the same category
      *      - e.g. meter to feet and inches --> all of them are length units.
      *
-     * @param inputUnit represents the source unit. (should be single or compound unit).
+     * @param targetUnit represents the source unit. (should be single or compound unit).
      * @param outputUnits represents the output unit. could be any type. (single, compound or mixed).
      * @param status
      */
-    ComplexUnitsConverter(const MeasureUnitImpl &inputUnit, const MeasureUnitImpl &outputUnits,
+    ComplexUnitsConverter(const MeasureUnitImpl &targetUnit, const MeasureUnitImpl &outputUnits,
```

Comment:
I think renaming this inputUnit to "targetUnit" doesn't make sense: we still have outputUnits here. Leave it as "inputUnit"?

Reply:
Missed one? "targetUnit represents the source unit" - source vs target.